src/other/index.jl

```diff
@@ -46,14 +46,41 @@ function names!(x::Index, nms::Vector{Symbol}; makeunique::Bool=false)
 end
 
 function rename!(x::Index, nms)
+    xbackup = copy(x)
+    processedfrom = Set{Symbol}()
+    processedto = Set{Symbol}()
+    toholder = Dict{Symbol,Int}()
     for (from, to) in nms
+        if from ∈ processedfrom
+            merge!(empty!(x.lookup), xbackup.lookup)
```

Comment:
Isn't this just `copy!(x.lookup, xbackup.lookup)`? (Use `Future.copy!` on Julia 1.0.)

Or maybe even simpler: outside of the loop, do
```julia
   if !allunique(keys(nms))
...
   if !allunique(values(nms))
```

Reply:
Yes it is exactly `copy!`. 

```julia
julia> nms = [:a=>:b, :a=>:c, :b=>:c]
3-element Array{Pair{Symbol,Symbol},1}:
 :a => :b
 :a => :c
 :b => :c

julia> allunique(keys(nms))
true
```

Reply:
Yeah replace `keys(nms)` with `first.(nms)` (or `(first(x) for x in nms)` to avoid an allocation).

Reply:
Implementation-wise, [`allunique`](https://github.com/JuliaLang/julia/blob/c6da87ff4bc7a855e217856757ad3413cf6d1f79/base/set.jl#L365-L373) is the same as here. 
However, we can get exact items that make things error by doing the check manually.

Reply:
Only `(first(x) for x in nms)` as broadcasting over dictionaries is not suported

Reply:
Ah, indeed. Though please use `copy!` at least, that will make the code simpler in the long run.

Reply:
I added a line of note on the `merge! ∘ empty!` to avoid adding a new dependency of `Future` for this package.
Since the package is not specially designed for 1.0, the `Future.blah` thing looks strange for people of 1.1 or after.

Reply:
`Future` is in the stdlib, so that's OK to have it as a dependency. And `Future.copy!` is much easier to understand to people used to Julia >= 1.1 than `merge!(empty(...), ...)`.